src/chessboard/components/SelectPromotionDialog.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { useChessboard } from "../context/chessboard-context";
+
+export const SelectPromotionDialog = ({
+  onChange,
+  popupCoords,
+}: {
+  onChange: any;
+  popupCoords: any;
+}) => {
+  const promotionoptions = ["q", "r", "n", "b"];
+  const {
+    boardWidth,
+
+    chessPieces,
+    promotion: { color = "w" },
+  } = useChessboard();
+  console.log({ color });
+  return (
+    <div
+      style={{
+        position: "absolute",
+        display: "grid",
+        width: "100px",
+        height: "100px",
+        top: `${popupCoords.y}px`,
+        left: `${popupCoords.x}px`,
+        zIndex: 100,
+        gridTemplateColumns: "1fr 1fr",
+        transform: "translate(-50px, -50px)",
+      }}
+    >
+      {promotionoptions.map((option) => (
+        <div
+          style={optionStyles(boardWidth)}
+          key={option}
+          onClick={() => onChange(option)}
```

Comment:
onChange is quite generic here, can I suggest a more descriptive name like `handlePromotion`